### ISSUE TITLE
:bulb: Improve stream_bufsize for stream.c:open_stream_out

### ISSUE THREAD

**Issue by pete4abw** [pete4abw/lrzip-next#47]
### lrzip-next Version

0.8.5

### Feature Suggestion

The `open_stream_out()` function sets the optimum segment size per chunk. For **zpaq**, it cannot be larger than the zpaq block size itself - 4096 (see #46 ). But for **lzma** the segment size is too small.If `control->overhead` x `number of threads` is less than the limit imposed by `control->usable_ram/testbufs`. Then the overhead should be the maximum segment size per thread. It's already computed. So, the solution resides in this block.

```
1095                 if (ZPAQ_COMPRESS && (limit/control->threads > 0x100000<<control->zpaq_bs))
1096                         stream_bufsize = round_up_page(control, MAX((0x100000<<control->zpaq_bs)-0x1000, STREAM_BUFSIZE));
1097                 else
1098                         stream_bufsize = round_up_page(control, MAX(limit/control->threads, STREAM_BUFSIZE));
1099
1100                 if (control->threads > 1)
1101                         print_maxverbose("Using up to %'d threads to compress up to %'"PRId64" bytes each.\n",
1102                                 control->threads, stream_bufsize);
1103                 else
1104                         print_maxverbose("Using only 1 thread to compress up to %'"PRId64" bytes\n",
1105                                 stream_bufsize);
```

So some work is ongoing at line **1097** to compute there can be a specific **lzma** adjustment for maximum segment size. The thought is that `control->overhead` should be the actual segment size per thread since it is already computed.

More later

### Steps to reproduce

N/A

### Relevant log output

```shell
N/A
```

### Please provide system details

N/A

### Additional Context

Work on `open_stream_out()` has been one of the earliest motivators for `lrzip-next`. Maximizing its effectiveness has been a priority. See [Wiki Article](https://github.com/pete4abw/lrzip-next/wiki/Chunk-Block-Size-vs-Number-of-Threads).

**Comment by pete4abw** [comment]
## Discussion when LZMA or ZPAQ used:
**Ramsize** = total system ram 
**Usable ram** = Ram size / 3 or 6 depending on whether or not STDIN/OUT is used
**Stream out Limit** = Usable ram / 2 (when compression is used)
**Overhead** = ram used by LZMA or ZPAQ (only applicable when LZMA or ZPAQ used)
**Threads * Overhead** has to be less than Limit

Instead of reducing Limit, we reduce threads and then overhead size.

Stream bufffer size should be Threads * Overhead, not Limit / Threads.

The goal is to have the largest possible buffer passed to backend.

**Comment by pete4abw** [comment]
When lzma levels are low, the dictionary size is really low, so that passing such a small buffer is impractical. For example, at Level 1, the lzma dictSize is only 256KB. At Level 3, just 4MB. So, control->overhead just won't do it.  Now, a focus will change to

`MAX(limit/threads, control->overhead)`

Work continues...

The same may be true with ZPAQ.

**Comment by pete4abw** [comment]
All done. Merged into [lzma-sdk-21.04beta](https://github.com/pete4abw/lrzip-next/tree/lzma-21.04beta).
